### PR TITLE
feat(skills): add edge case hunter as parallel review layer in PR review

### PR DESCRIPTION
**Pull request by alexeyv**
## Summary

- Adds the edge case hunter (`review-edge-case-hunter.xml`) as a second review layer in `bmad-os-review-pr`, running in parallel with the adversarial review as concurrent subagents
- After both complete, findings are merged, deduplicated (keeping the more specific version), and tagged by source (`[Adversarial]`, `[Edge Case]`, `[Both]`)
- Tone transformation applies to the merged set; output format now shows source attribution and per-layer counts
- Removes obsolete "Ultrathink" keyword from cynical review prompt
- Fixes contradictions in edge case hunter task spec from #1790 review feedback (output format array example, zero-findings halt, empty-content handling, step execution wording)

## Design

The two layers are intentionally orthogonal:
- **Adversarial** (step 1.1): attitude-driven — cynical persona, finds issues through skepticism
- **Edge Case Hunter** (step 1.2): method-driven — exhaustive path walking, reports only unhandled gaps

Running them as parallel subagents keeps latency ~same as before (single-layer review).

## Test plan

- [x] Verify all existing tests pass (`npm test`)
- [x] Run the PR review skill on a real PR and confirm both layers produce findings
- [x] Confirm deduplication correctly merges overlapping findings
- [ ] Confirm source tags appear in final output

### CODE REVIEW

File: .claude/skills/bmad-os-review-pr/prompts/instructions.md [hunk 1]
```diff
@@ -93,13 +93,17 @@ gh pr diff {PR_NUMBER} [--repo {REPO}] --name-only | grep -E '\.(png|jpg|jpeg|gi
 
 Store list of binary files to skip. Note them in final output.
 
-## Adversarial Review
+## Review Layers
 
-### 1.1 Run Cynical Review
+**Launch steps 1.1 and 1.2 as parallel subagents** using the Task tool. Both receive the same PR diff and run concurrently. Wait for both to complete before proceeding to step 1.3.
```
Comment (bmadcode) [inline .claude/skills/bmad-os-review-pr/prompts/instructions.md]: do we need to say using the task agents? ive found just telling it subagents works and will give better chance of also working with any tool that uses subagents. The task tool itself will probably block codex,

**Review by bmadcode — CHANGES_REQUESTED** [review]

File: .claude/skills/bmad-os-review-pr/prompts/instructions.md [hunk 2]
```diff
@@ -124,14 +128,64 @@ Likely tag:
...
+
+```markdown
+### [NUMBER]. [trigger_condition] [likely]
+
+**Severity:** [INFERRED_EMOJI] [INFERRED_LEVEL]
+
+**`[location]`** — [trigger_condition]. [potential_consequence].
+
+**Suggested fix:**
+```
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply .claude/skills/bmad-os-review-pr/prompts/instructions.md:147]: fixed

File: src/core/tasks/review-edge-case-hunter.xml [hunk 3]
```diff
@@ -15,16 +15,16 @@ Ignore the rest of the codebase unless the provided content explicitly reference
   </inputs>
 
   <output-format>Return ONLY a valid JSON array of objects. Each object must contain exactly these four fields and nothing else:
-{
+[{
   "location": "file:line",
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: noise

File: src/core/tasks/review-edge-case-hunter.xml [hunk 4]
```diff
@@ -15,16 +15,16 @@ Ignore the rest of the codebase unless the provided content explicitly reference
...
   "trigger_condition": "one-line description (max 15 words)",
   "guard_snippet": "minimal code sketch that closes the gap",
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: noise

File: src/core/tasks/review-edge-case-hunter.xml [hunk 5]
```diff
@@ -39,7 +39,7 @@ No extra text, no explanations, no markdown wrapping.</output-format>
   <flow>
     <step n="1" title="Receive Content">
       <action>Load the content to review from provided input or context</action>
-      <action>If content to review is empty, ask for clarification and abort task</action>
+      <action>If content to review is empty or unreadable, HALT per halt-conditions</action>
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: fixed

File: .claude/skills/bmad-os-review-pr/prompts/instructions.md (hunk 1 above)
Comment (alexeyv) [reply .claude/skills/bmad-os-review-pr/prompts/instructions.md]: No, we don't.

File: src/core/tasks/review-edge-case-hunter.xml [hunk 6]
```diff
@@ -57,7 +57,7 @@ No extra text, no explanations, no markdown wrapping.</output-format>
   </flow>
 
   <halt-conditions>
-    <condition>HALT if zero findings - this is suspicious, re-analyze or ask for guidance</condition>
+    <condition>If zero findings after initial analysis, re-analyze once with increased scrutiny. If still zero, return empty array [].</condition>
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: noise

File: src/core/tasks/review-edge-case-hunter.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: noise

File: src/core/tasks/review-edge-case-hunter.xml [hunk 7]
```diff
@@ -15,16 +15,16 @@ Ignore the rest of the codebase unless the provided content explicitly reference
...
-{
+[{
   "location": "file:line",
   "trigger_condition": "one-line description (max 15 words)",
   "guard_snippet": "minimal code sketch that closes the gap",
   "potential_consequence": "what could actually go wrong (max 15 words)"
-}
-No extra text, no explanations, no markdown wrapping.</output-format>
+}]
+No extra text, no explanations, no markdown wrapping. An empty array [] is valid when no unhandled paths are found.</output-format>
```
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml:24]: fixed

File: src/core/tasks/review-edge-case-hunter.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (alexeyv) [reply src/core/tasks/review-edge-case-hunter.xml]: fixed

File: .claude/skills/bmad-os-review-pr/prompts/instructions.md (hunk 1 above)
Comment (alexeyv) [reply .claude/skills/bmad-os-review-pr/prompts/instructions.md]: fixed
